sound/soc/sof/loader.c

```diff
@@ -61,11 +60,13 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 		/* process structure data */
 		switch (ext_hdr->type) {
 		case SOF_IPC_EXT_DMA_BUFFER:
+			ret = 0;
 			break;
 		case SOF_IPC_EXT_WINDOW:
 			ret = get_ext_windows(sdev, ext_hdr);
 			break;
 		default:
+			ret = 0;
 			break;
 		}
```

Comment:
please fix the conflict and the commit message (using 'there is' is too vague, use more active wording)

An error occurs during parsing more than one ext_data from the mailbox, because of invalid data offset handling. Fix by removing the incorrect duplicate increment of the offset.

The return value is also reset in the switch case. This does not change the behavior but improves readability - there is no longer a need to check what the return value of get_ext_windows is.

Reply:
note as @dbaluta suggested that the second part should be in a separate patch, it's not logically tied to the offset change.